src/include/optimizer/rule_impls.h

```diff
@@ -19,29 +19,37 @@
 namespace peloton {
 namespace optimizer {
 
```

Comment:
You need to comment on what each of these rules does. Just the name is not enough.